### PR TITLE
Bug 1206306 - Migrate remaining mozL10n.DateTimeFormat's in Communications to Intl API

### PR DESCRIPTION
**Pull request by zbraniecki**
https://bugzilla.mozilla.org/show_bug.cgi?id=1206306

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=accb0aaab2bd94bdfe561b58d0707720156b6bc9

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=fd4c2d2c0e9850b5dd247a9f931644bbacb07fd2

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=5529de04deeff8b000841023c18a11446dcb7610

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=de051b6552c01e8836027429b34c9d69f6cf277a

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=43bdb2873903891401d0d4900a640f4b3e6bc5c7

File: apps/communications/contacts/test/unit/views/settings_test.js [hunk 1]
```diff
@@ -501,8 +501,14 @@ suite('Contacts settings >', function() {
         assert.equal(time.getAttribute('datetime'),
             (new Date(timestamps[source])).toLocaleString());
         assert.equal(time.textContent, utils.time.pretty(timestamps[source]));
+        var timeString = new Date(timestamps[source]).toLocaleString(
+          navigator.languages, {
+            hour12: navigator.mozHour12,
+            hour: 'numeric',
+            minute: 'numeric',
+          });
         if (extraString) {
```
Comment (DouglasSherk) [inline apps/communications/contacts/test/unit/views/settings_test.js]: This isn't a good name for the parameter anymore.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=4e6fb46dc530f3dd65b039a6f49a05cdbdfc5254

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=cd7d3ad9f2ebd4982e568e4627312227e5c6eb35

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=4c4009571e51ef5bfcf4eccd04b25a37d78ce753

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=c1e6677a0f621c110146b13a4189660b001f1b80

**Pull request merged**